app/models/remote_taxons.rb

```diff
@@ -1,6 +1,7 @@
+# TODO: Split this up and move it into the BulkTagging & Taxonomy namespaces.
```

Comment:
Which part of this would live in `BulkTagging`? This seems like it could all go in `Taxonomy` to me, but I'm not familiar with how it's modelled.

It seems like `#parents_for_taxon` is something you would want to call on `Taxon`, and taxon searching could also live in the taxonomy namespace.